include/pisa/recursive_graph_bisection.hpp

```diff
@@ -226,18 +245,31 @@ void swap(document_partition<Iterator>& partition, degree_map_pair& degrees)
 }
 
 template <class Iterator, class GainF>
-void process_partition(document_partition<Iterator>& partition, GainF gain_function, int iterations = 20)
+void process_partition(
+    document_partition<Iterator>& partition,
+    GainF gain_function,
+    bp::ThreadLocal& thread_local_data,
+    int iterations = 20)
 {
-    thread_local single_init_vector<size_t> left_degree(partition.left.term_count());
-    left_degree.clear();
-    thread_local single_init_vector<size_t> right_degree(partition.right.term_count());
-    right_degree.clear();
+    bool exists = false;
+    auto& left_degree = thread_local_data.left_degrees.local(exists);
+    if (exists) {
+        left_degree.clear();
+    } else {
+        left_degree.resize(partition.left.term_count());
+    }
+    auto right_degree = thread_local_data.right_degrees.local(exists);
+    if (exists) {
```

Comment:
Wouldn't be better to use the same logic as above: `if(not exist)`
Also can we have a function for this logic?